src/lib/inputs/InputNumber.svelte

```diff
@@ -0,0 +1,397 @@
+<script lang="ts">
+    import { createEventDispatcher, onMount } from "svelte";
+
+    // Props
+    export let variant: "default" | "line" | "outline" = "default";
+    export let size: "xsmall" | "small" | "medium" | "large" | "xlarge" =
+        "medium";
+    export let background: string = "white";
+    export let color: string = "#000";
+    export let value: number = 0;
+    export let style: string = "";
+    export let label: boolean = false;
+    export let labelColor: string = "#000";
+    export let disabled: boolean = false;
+    export let isError: boolean = false;
+    export let isLoading: boolean = false;
+    export let min: number = -10000;
+    export let max: number = 10000;
+    export let step: number | "any" = 1;
+
+    // Event dispatcher
+    const dispatch = createEventDispatcher();
+
+    // Variables
+    let containerClassList = ["input"];
+    let containerClassString = "";
+    let readOnly: boolean = false;
+
+    // Lifecycle Hooks
+    onMount(() => {
+        containerClassList.push(size);
+        containerClassList.push(variant);
+        if (isLoading) {
+            containerClassList.push("loading");
+        } else if (isError) {
+            containerClassList.push("error");
+        }
+        containerClassString = containerClassList.join(" ");
+
+        if (value < min) {
+            value = min;
+        } else if (value > max) {
+            value = max;
+        }
```

Comment:
would this be needed in the event the developer set an initial value higher than what they set as the max value or lower than what they set as  the min value themselves?

Reply:
Oh, interesting... Just tested this an regardless of initial value, the min or max take precedence over initial value. Which I think is how it should be. Unless you disagree for whatever reason.